t/apicast-async-reporting.t

```diff
@@ -1,202 +1,181 @@
-lua_shared_dict api_keys 1m;
-lua_package_path "$TEST_NGINX_LUA_PATH";
-init_by_lua_block {
-  ngx.shared.api_keys:set('42:value:usage%5Bhits%5D=2', 200)
-  require('apicast.configuration_loader').mock({
-    services = {
+--- configuration
+{
+    "services": [
       {
-        id = 42,
-        backend_version = 1,
-        backend_authentication_type = 'service_token',
-        backend_authentication_value = 'token-value',
-        proxy = {
-          backend = { endpoint = "http://127.0.0.1:$TEST_NGINX_SERVER_PORT" },
-          api_backend = "http://127.0.0.1:$TEST_NGINX_SERVER_PORT/api-backend/",
-          proxy_rules = {
-            { pattern = '/', http_method = 'GET', metric_system_name = 'hits', delta = 2 }
-          }
+        "id": 42,
+        "backend_version": 1,
+        "backend_authentication_type": "service_token",
+        "backend_authentication_value": "token-value",
+        "proxy": {
+            "api_backend": "http://test:$TEST_NGINX_SERVER_PORT/api-backend/",
+            "proxy_rules": [
+                { "pattern" : "/", "http_method" : "GET", "metric_system_name" : "hits", "delta" : 2 }
+            ]
         }
       }
-    }
-  })
+    ]
 }
---- config
-include $TEST_NGINX_APICAST_CONFIG;
-
+--- backend
 location /transactions/authrep.xml {
   content_by_lua_block {
     local expected = "service_token=token-value&service_id=42&usage%5Bhits%5D=2&user_key=value"
     require('luassert').same(ngx.decode_args(expected), ngx.req.get_uri_args(0))
   }
 }
-
+--- upstream
 location /api-backend/ {
   echo 'yay, api backend: $http_host';
 }
---- request
-GET /?user_key=value
---- response_body env
-yay, api backend: 127.0.0.1:$TEST_NGINX_SERVER_PORT
---- error_code: 200
+--- pipelined_requests eval
+["GET /?user_key=value","GET /?user_key=value"]
+--- response_body env eval
+["yay, api backend: test:$TEST_NGINX_SERVER_PORT\x{0a}","yay, api backend: test:$TEST_NGINX_SERVER_PORT\x{0a}"]
+--- error_code eval
+["200","200"]
 --- no_error_log
 [error]
 
+
 === TEST 2: https api backend works
 with async background reporting
 --- ssl random_port
---- http_config
-include $TEST_NGINX_UPSTREAM_CONFIG;
-lua_package_path "$TEST_NGINX_LUA_PATH";
-init_by_lua_block {
-  ngx.shared.api_keys:set('42:foo:usage%5Bhits%5D=1', 200)
-  require('apicast.configuration_loader').mock({
-    services = {
-      {
-        id = 42,
-        backend_version = 1,
-        backend_version = 1,
-        backend_authentication_type = 'service_token',
-        backend_authentication_value = 'token-value',
-        proxy = {
-          backend = { endpoint = "https://127.0.0.1:$TEST_NGINX_RANDOM_PORT" },
-          api_backend = "http://127.0.0.1:$TEST_NGINX_SERVER_PORT/api/",
-          proxy_rules = {
-            { pattern = '/', http_method = 'GET', metric_system_name = 'hits', delta = 1 }
-          }
-        }
+--- env random_port eval
+(
+  'BACKEND_ENDPOINT_OVERRIDE' => "https://test_backend:$ENV{TEST_NGINX_RANDOM_PORT}"
```

Comment:
Is this actually needed? Check https://github.com/3scale/Test-APIcast/blob/c3e059369795f76156cbc7fdd6076ad2fcc1a5c7/lib/Test/APIcast/Blackbox.pm#L86